### PR TITLE
Implement Sendable

### PR DESCRIPTION
**Pull request by Davidde94**
Make every class `Sendable` (or `@unchecked Sendable`) to prepare the way for full async/await support in NIO.

### CODE REVIEW

File: json-rpc/Sources/JsonRpc/Codec.swift [hunk 1]
```diff
@@ -94,7 +97,7 @@ internal final class NewlineEncoder: ByteToMessageDecoder, MessageToByteEncoder
 // 1 byte: a colon (":", 0x3a), not included in LEN
 // LEN bytes: a JSON/RPC message, no leading or trailing whitespace
 // 1 byte: a newline (0x0a), not included in LEN
-internal final class JSONPosCodec: ByteToMessageDecoder, MessageToByteEncoder {
+internal final class JSONPosCodec: ByteToMessageDecoder, MessageToByteEncoder, Sendable {
```
Comment (Lukasa) [inline json-rpc/Sources/JsonRpc/Codec.swift:100]: This is a `ChannelHandler` with no other API so shouldn't be `Sendable`.

File: json-rpc/Sources/JsonRpc/Codec.swift [hunk 2]
```diff
@@ -161,14 +164,14 @@ internal final class JSONPosCodec: ByteToMessageDecoder, MessageToByteEncoder {
 }
 
 // no delimeter is provided, brute force try to decode the json
-internal final class BruteForceCodec<T>: ByteToMessageDecoder, MessageToByteEncoder where T: Decodable {
+internal final class BruteForceCodec<T>: ByteToMessageDecoder, MessageToByteEncoder, @unchecked Sendable where T: Decodable {
```
Comment (Lukasa) [inline json-rpc/Sources/JsonRpc/Codec.swift:167]: This is a `ChannelHandler` with no other API so shouldn't be `Sendable`.

File: json-rpc/Sources/JsonRpc/Codec.swift [hunk 3]
```diff
@@ -217,7 +222,7 @@ internal final class BruteForceCodec<T>: ByteToMessageDecoder, MessageToByteEnco
 }
 
 // bytes to codable and back
-internal final class CodableCodec<In, Out>: ChannelInboundHandler, ChannelOutboundHandler where In: Decodable, Out: Encodable {
+internal final class CodableCodec<In, Out>: ChannelInboundHandler, ChannelOutboundHandler, Sendable where In: Decodable, Out: Encodable {
```
Comment (Lukasa) [inline json-rpc/Sources/JsonRpc/Codec.swift:225]: This is a `ChannelHandler`, so it shouldn't be `Sendable`.

File: json-rpc/Sources/JsonRpc/Server.swift [hunk 4]
```diff
@@ -1,11 +1,18 @@
...
 
-public final class TCPServer {
+public final class TCPServer: @unchecked Sendable {
     private let group: MultiThreadedEventLoopGroup
     private let config: Config
     private var channel: Channel?
     private let closure: RPCClosure
+    private var state = State.initializing {
+        didSet {
+            print("\(self) \(state)")
```
Comment (Lukasa) [inline json-rpc/Sources/JsonRpc/Server.swift:12]: We should probably remove the `didSet`.

File: json-rpc/Sources/JsonRpc/Server.swift [hunk 5]
```diff
@@ -93,7 +90,7 @@ public final class TCPServer {
     }
 }
 
-private class Handler: ChannelInboundHandler {
+private class Handler: ChannelInboundHandler, Sendable {
```
Comment (Lukasa) [inline json-rpc/Sources/JsonRpc/Server.swift:93]: This is a `ChannelHandler`, so it shouldn't be `Sendable`.

File: nio-launchd/Sources/nio-launchd/Client.swift [hunk 6]
```diff
@@ -32,28 +33,33 @@ struct Client: ParsableCommand {
     }
 }
 
-private final class EchoHandler: ChannelInboundHandler {
+private final class EchoHandler: ChannelInboundHandler, @unchecked Sendable {
```
Comment (Lukasa) [inline nio-launchd/Sources/nio-launchd/Client.swift:36]: This is a `ChannelHandler` so it shouldn't be `Sendable`.

**Review by Lukasa — COMMENTED** [review]
Thanks for this! I've left some notes in the diff.

File: TLSify/Sources/TLSifyLib/GlueHandler.swift [hunk 7]
```diff
@@ -14,8 +14,10 @@
 
 import NIO
 import Logging
+import NIOConcurrencyHelpers
```
Comment (Lukasa) [inline TLSify/Sources/TLSifyLib/GlueHandler.swift:17]: We don't need this import anymore.

File: TLSify/Sources/TLSifyLib/TLSProxy.swift [hunk 8]
```diff
@@ -15,6 +15,7 @@
 import NIO
 import NIOSSL
 import Logging
+import NIOConcurrencyHelpers
```
Comment (Lukasa) [inline TLSify/Sources/TLSifyLib/TLSProxy.swift:18]: We don't need this import anymore.

File: backpressure-file-io-channel/Sources/BackpressureChannelToFileIO/SaveEverythingHTTPServer.swift [hunk 9]
```diff
@@ -15,6 +15,7 @@
 import NIO
 import NIOHTTP1
 import Logging
+import NIOConcurrencyHelpers
```
Comment (Lukasa) [inline backpressure-file-io-channel/Sources/BackpressureChannelToFileIO/SaveEverythingHTTPServer.swift:18]: We don't need this import anymore.

File: http2-client/Sources/http2-client/main.swift [hunk 10]
```diff
@@ -19,14 +19,15 @@ import NIOTLS
 import NIOSSL
 import Foundation
 import NIOExtras
+import NIOConcurrencyHelpers
```
Comment (Lukasa) [inline http2-client/Sources/http2-client/main.swift:22]: We don't need this import anymore.

File: json-rpc/Sources/JsonRpc/Codec.swift [hunk 11]
```diff
@@ -1,6 +1,7 @@
 import Foundation
 import NIO
 import NIOFoundationCompat
+import NIOConcurrencyHelpers
```
Comment (Lukasa) [inline json-rpc/Sources/JsonRpc/Codec.swift:4]: We don't need this import anymore.

File: nio-launchd/Sources/nio-launchd/Client.swift [hunk 12]
```diff
@@ -14,6 +14,7 @@
 
 import NIO
 import ArgumentParser
+import NIOConcurrencyHelpers
```
Comment (Lukasa) [inline nio-launchd/Sources/nio-launchd/Client.swift:17]: We don't need this import anymore.

File: nio-launchd/Sources/nio-launchd/Server.swift [hunk 13]
```diff
@@ -15,6 +15,7 @@
 import NIO
 import ArgumentParser
 import launch
+import NIOConcurrencyHelpers
```
Comment (Lukasa) [inline nio-launchd/Sources/nio-launchd/Server.swift:18]: We don't need this import anymore.

**Review by Lukasa — APPROVED** [review]
If you remove the remaining unnecessary imports we're good to go.

**Comment by Davidde94** [comment]
@Lukasa thanks for the reviews, everything's sorted

**Pull request merged**